src/components/ProductBlock.tsx

```diff
@@ -41,71 +38,76 @@ interface IProps extends Partial<RouteComponentProps<MatchParams>>, WrappedCompo
     subscriptionId?: string;
 }
 
-interface IState {
-    errors: Partial<Record<keyof iProductBlock, boolean>>;
-    required: Column[];
-    duplicateName: boolean;
-    initial: boolean;
-    isNew: boolean;
-    productBlock?: iProductBlock;
-    processing: boolean;
-    resourceTypes: ResourceType[];
-    productBlocks: iProductBlock[];
-    showConfirmDialog: ShowConfirmDialogType;
-    closeConfirmDialog: () => void;
-}
-
-class ProductBlock extends React.Component<IProps, IState> {
-    static contextType = ApplicationContext;
-    context!: React.ContextType<typeof ApplicationContext>;
-    state: IState = {
-        errors: {},
-        required: ["name", "description"],
-        duplicateName: false,
-        initial: true,
-        isNew: true,
-        processing: false,
-        resourceTypes: [],
-        productBlocks: [],
-        showConfirmDialog: () => {},
-        closeConfirmDialog: () => {},
-    };
+function ProductBlock({ match }: IProps) {
+    const { apiClient, redirect, allowed } = useContext(ApplicationContext);
+    const { showConfirmDialog, closeConfirmDialog } = useContext(ConfirmationDialogContext);
+    const [errors, setErrors] = useState<Partial<Record<keyof iProductBlock, boolean>>>({});
+    const [required] = useState<Column[]>(["name", "description"]);
+    const [duplicateName, setDuplicateName] = useState<boolean>(false);
+    const [initial, setInitial] = useState<boolean>(true);
+    const [productBlock, setProductBlock] = useState<iProductBlock>({
+        created_at: 0,
+        description: "",
+        end_date: 0,
+        name: "",
+        parent_ids: [],
+        product_block_id: "",
+        resource_types: [],
+        status: "",
+        tag: "",
+    });
+    const [processing, setProcessing] = useState<boolean>(false);
+    const [productBlocks, setProductBlocks] = useState<iProductBlock[]>([]);
+    const [readOnly, setReadOnly] = useState<boolean>(false);
+    const [productBlockLoaded, setProductBlockLoaded] = useState(false);
 
-    componentDidMount() {
-        const id = this.props.match?.params.id;
-        if (id !== "new") {
-            this.context.apiClient
-                .productBlockById(id!)
-                .then((res: iProductBlock) => this.setState({ productBlock: res, isNew: false }));
+    useEffect(() => {
+        const product_block_id = match?.params.id;
+        if (product_block_id !== "new") {
+            apiClient.productBlockById(product_block_id!).then((res: iProductBlock) => {
+                setProductBlock(res);
+                setReadOnly(!allowed("/orchestrator/metadata/product/edit/" + productBlock.product_block_id + "/"));
```

Comment:
Looks like this should be:

```suggestion
                setReadOnly(!allowed("/orchestrator/metadata/product-block/edit/" + productBlock.product_block_id + "/"));
```